software/image_processor.py

```diff
@@ -0,0 +1,263 @@
+        return "[Object: x={}; y={}; size={}; distance={}; exists={}]".format(self.x, self.y, self.size, self.distance, self.exists)
+
+
+# results object of image processing. contains coordinates of objects and frame data used for these results
+class ProcessedResults():
+    def __init__(self, 
+                balls=[], 
+                basket_b = Object(exists = False), 
+                basket_m = Object(exists = False), 
+                color_frame = [],
+                depth_frame = [],
+                fragmented = [],
+                debug_frame = []) -> None:
+
+        self.balls = balls
+        self.basket_b = basket_b
+        self.basket_m = basket_m
+        self.color_frame = color_frame
+        self.depth_frame = depth_frame
+        self.fragmented = fragmented
+
+        # can be used to illustrate things in a separate frame buffer
+        self.debug_frame = debug_frame
+
+
+#Main processor class. processes segmented information
+class ImageProcessor():
+    def __init__(self, camera, color_config = "colors/colors.pkl", debug = False):
+        self.camera = camera
+        self.kernelSmall = cv2.getStructuringElement(cv2.MORPH_RECT, (2,2))
+        self.kernelLarge = cv2.getStructuringElement(cv2.MORPH_RECT, (5,5))
+        self.color_config = color_config
+        with open(self.color_config, 'rb') as conf:
+            self.colors_lookup = pickle.load(conf)
+            self.set_segmentation_table(self.colors_lookup)
+
+        #each of these is just an array of ints, which correspond to colors
+        self.fragmented	= np.zeros((self.camera.rgb_height, self.camera.rgb_width), dtype=np.uint8)
+        self.t_balls = np.zeros((self.camera.rgb_height, self.camera.rgb_width), dtype=np.uint8)
+        self.t_basket_b = np.zeros((self.camera.rgb_height, self.camera.rgb_width), dtype=np.uint8)
+        self.t_basket_m = np.zeros((self.camera.rgb_height, self.camera.rgb_width), dtype=np.uint8)
+
+        self.debug = debug
+        self.debug_frame = np.zeros((self.camera.rgb_height, self.camera.rgb_width), dtype=np.uint8)
+
+    def set_segmentation_table(self, table):
+        segment.set_table(table)
+
+    def start(self):
+        self.camera.open()
+
+    def stop(self):
+        self.camera.close()
+
+    def find_multiple_matching_pixels(self,color, x, y, image, direction, pixels): 
+            pixel_counter = pixels
+            try:
+                while pixel_counter > 0:
+                    if image[x, y] == color:
+                        pixel_counter -= 1
+                    else:
+                        pixel_counter = 10
+                    if direction == "-y":
+                        y -= 1
+                    if direction == "y":
+                        y += 1
+                    if direction == "-x":
+                        x -= 1
+                    if direction == "x":
+                        x += 1
+                return x, y
+            except:
+                return False
```

Comment:
You currently return different data based on a methods exit conditions. This is error prone

Reply:
@King-Of-Nothing25

Reply:
I removed this code, we didn't use it